src/stream/src/common/table/state_table.rs

```diff
@@ -1010,8 +1013,10 @@ impl<S: StateStore> StateTable<S> {
 
         // Construct prefix hint for prefix bloom filter.
         let pk_prefix_indices = &self.pk_indices[..pk_prefix.len()];
-        let prefix_hint = {
-            if self.dist_key_indices.is_empty() || self.dist_key_indices != pk_prefix_indices {
+        let dist_key_hint = {
+            if self.dist_key_indices.is_empty()
+                || !is_subset(self.dist_key_indices.clone(), pk_prefix_indices.to_vec())
+            {
                 None
             } else {
                 Some([&vnode, &encoded_prefix[..]].concat())
```

Comment:
We are still using the full prefix key as the dist_key_hint here, which is incorrect when dist_key is no longer the full prefix.

Reply:
Same issue for the `StorageTable` change.